R/style_scale_fill.R

```diff
@@ -33,21 +40,42 @@ reverse = F, ...){
 
 #################################
 
-#' Access Palettes
-#'
-#' @description Access any of the color palettes available
-#'
-#' @param palette Character name of palette in cllr_palettes
-#' @param reverse Boolean; Should the colors be reversed?
+#' Colors
 #'
+#' # this is from: coolors.co
+# https://www.color-hex.com/color-palette/28194
+# Add in grey scale
+# and color blind friendly
+colors <- c(
+  # duck hunt colors
+  `azure` = "#32b5fc",
+  `green` = "#75cb0b",
+  `brown` = "#7e6500",
+  `dark blue` = "#00009f",
+  `sunset` = "#fc6955",
+  # color blind colors
+  `magenta` = "#FF01FB",
+  `cerulean` = "#02A9EA",
+  `lemon glacier` = "#FAFF00",
+  `malachite` = "#20BF55",
+  `imp blue` = "#0B4F6C",
+  # gray scale colors
+  `black` = "#171800",
+  `light gray` = "#CFD2DC"
+)
 
+#################################
 
-cllr_pal <- function(palette = "main", reverse=F, ...){
-  pal <- cllr_palettes[[palette]]
+#' Grab any combination of colors for our palettes
+#'
+#' @param ... colors ya want
 
-  if (reverse) pal <- rev(pal)
+cllr_colors <- function(...){
+  cols <- c(...)
+  if (is.null(cols))
+    return(colors)
 
-  colorRampPalette(pal, ...)
+  colors[cols]
 }
```

Comment:
Random thought, but are these palettes actually data? Is there a reason to create them every time the function is called? I am really thinking about why we need to recreate them for every plot and keeping a single list of colors that is easy to update, in some ways they feel like data, internal data perhaps.